plugins/juju/04known_bugs

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/python3
+import logging
+import subprocess
+
+from common import helpers
+from common.constants import JUJU_LOG_PATH
+
+log = logging.getLogger("juju.04known-bugs")
+
+
+def detect_bug1910958():
+    """Unit fails to start complaining there are members in the relation."""
+    result = {"bug": "LP: #1910958",
+              "description": ("Unit fails to start complaining there are "
+                              "members in the relation."),
+              }
+    pattern = (
+        'manifold worker returned unexpected error: failed to initialize '
+        'uniter for "[A-Za-z0-9\\-]+": cannot create relation state tracker: '
+        'cannot remove persisted state, relation \\d+ has members'
+    )
+
+    cmd = ["grep", "-r", "-P", "-H", "--include=*.log", "--files-with-match",
```

Comment:
any attempt to implement `grep` in python would fail miserably[0]. I understand the desire to not be using subprocess around, I'm in that boat too, but I truly believe that hotsos is better served relying on gnu grep

[0] https://lists.freebsd.org/pipermail/freebsd-current/2010-August/019310.html

Reply:
I moved the code that calls grep to common.grep(). If this really needs to be reimplemented in pure python I can do that, I could do it too.

Reply:
We can use this for now if you want but what we really need is a solution that scales. Imagine a scenario where we want to search a set of files using a set of keys. Running grep over each file for each key is going to take a very long time (in my experience grep regex is really slow). With a little thought I'm sure we could come up with something quite powerful that is flexible enough to search for anything we need and can do it must faster than grep. For example as a starting point I quite like the solutions proposed in https://www.blopig.com/blog/2016/08/processing-large-files-using-python/. That combined with a search classifier would be great.

Reply:
I've had a go at implementing some search tools, please take a look and let me know what you think - https://github.com/dosaboy/hotsos/pull/50